src/Controller/Controller.php

```diff
@@ -282,30 +279,20 @@ public function loadComponent(string $name, array $config = []): Component
     }
 
     /**
-     * Magic accessor for model autoloading.
+     * Magic accessor to provide info about model auto loading removal.
      *
      * @param string $name Property name
-     * @return \Cake\Datasource\RepositoryInterface|null The model instance or null
+     * @return null
      */
-    public function __get(string $name): ?RepositoryInterface
+    public function __get(string $name)
     {
-        if (!empty($this->modelClass)) {
-            if (!str_contains($this->modelClass, '\\')) {
-                [, $class] = pluginSplit($this->modelClass, true);
-            } else {
-                $class = App::shortName($this->modelClass, 'Model/Table', 'Table');
-            }
-
-            if ($class === $name) {
-                return $this->loadModel();
-            }
-        }
-
         $trace = debug_backtrace();
         $parts = explode('\\', static::class);
         trigger_error(
             sprintf(
-                'Undefined property: %s::$%s in %s on line %s',
+                'Undefined property: %s::$%s in %s on line %s.'
+                    . 'Auto loading of default model as property has been removed.'
+                    . ' Use $this->getTable() instead.',
```

Comment:
This message is a little awkward. Maybe `The default model is no longer auto-loaded and assigned to a property.`